File: backend/rename_ysb_to_ybs.py
import os
import re

def rename_in_file(file_path):
    with open(file_path, 'r', encoding='utf-8') as file:
        content = file.read()
    
    # Replace YBS with YBS, being careful about quotes and word boundaries
    modified = re.sub(r'\bYSB\b', 'YBS', content)
    modified = re.sub(r"'YBS'", "'YBS'", modified)
    modified = re.sub(r'"YBS"', '"YBS"', modified)
    modified = re.sub(r'/ybs/', '/ybs/', modified)
    modified = re.sub(r'\bysb\b', 'ybs', modified)
    
    # Write back if changes were made
    if content != modified:
        with open(file_path, 'w', encoding='utf-8') as file:
            file.write(modified)
        print(f"Updated: {file_path}")

def walk_directory(directory):
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.endswith(('.py', '.html', '.js', '.jsx')):
                file_path = os.path.join(root, file)
                rename_in_file(file_path)

if __name__ == "__main__":
    backend_dir = '.'
    walk_directory(backend_dir)
    print("Backend renaming completed.")

File: frontend/rename-ysb-to-ybs.js
const fs = require('fs');
const path = require('path');

function renameInFile(filePath) {
  try {
    let content = fs.readFileSync(filePath, 'utf8');
    
    // Original content for comparison
    const originalContent = content;
    
    // Replace YSB with YBS, careful with different contexts
    content = content.replace(/\bYSB\b/g, 'YBS');
    content = content.replace(/\/ysb\//g, '/ybs/');
    content = content.replace(/\bysb\b/g, 'ybs');
    content = content.replace(/5YB/g, '5YB'); // Keep 5YB as is, don't change to 5YS
    
    // Write back only if changes were made
    if (content !== originalContent) {
      fs.writeFileSync(filePath, content, 'utf8');
      console.log(`Updated: ${filePath}`);
    }
  } catch (error) {
    console.error(`Error processing ${filePath}: ${error.message}`);
  }
}

function walkDirectory(directory) {
  const files = fs.readdirSync(directory);
  
  files.forEach(file => {
    const filePath = path.join(directory, file);
    const stats = fs.statSync(filePath);
    
    if (stats.isDirectory() && !filePath.includes('node_modules') && !filePath.includes('.git')) {
      walkDirectory(filePath);
    } else if (stats.isFile() && 
              (filePath.endsWith('.js') || 
               filePath.endsWith('.jsx') || 
               filePath.endsWith('.html') || 
               filePath.endsWith('.css'))) {
      renameInFile(filePath);
    }
  });
}

walkDirectory('src');
console.log('Frontend renaming completed.');